### ISSUE TITLE
Not longer working with Valetudo 2021.01.1

### ISSUE THREAD

**Issue by sylvaindd** [Hypfer/ICantBelieveItsNotValetudo#45]
It doesn't seem to be working with the new Valetudo version 2021.01.1.

![image](https://user-images.githubusercontent.com/5629405/106021962-a18b0b00-60c5-11eb-9c8a-9a11f7feb4f1.png)

```
SyntaxError: Unexpected token e in JSON at position 0
    at JSON.parse (<anonymous>)
    at MqttClient.<anonymous> (/app/lib/MqttClient.js:86:42)
    at MqttClient.emit (events.js:315:20)
    at MqttClient._handlePublish (/app/node_modules/mqtt/lib/client.js:1277:12)
    at MqttClient._handlePacket (/app/node_modules/mqtt/lib/client.js:410:12)
    at work (/app/node_modules/mqtt/lib/client.js:321:12)
    at Writable.writable._write (/app/node_modules/mqtt/lib/client.js:335:5)
    at doWrite (/app/node_modules/mqtt/node_modules/readable-stream/lib/_stream_writable.js:409:139)
    at writeOrBuffer (/app/node_modules/mqtt/node_modules/readable-stream/lib/_stream_writable.js:398:5)
    at Writable.write (/app/node_modules/mqtt/node_modules/readable-stream/lib/_stream_writable.js:307:11)
```

Unfortunatly I removed my old topic, but I guess the map_data was previously in JSON as it tries to parse it as  JSON.

**Comment by Hypfer** [comment]
Yep the map data is deflated and base64 encoded (optional) now and needs to be inflated first

**Comment by sylvaindd** [comment]
I made it work without base64 encoding.
```
zlib.inflate(message, (err, result) => {
  if(!err) {
    this.updateMapTopic(JSON.parse(result));
  } else {
    console.error(err);
  }
});
```
Tried this for base64 encoded, but didn't work :
```
zlib.inflate(Buffer.from(message, 'base64'), (err, result) => {
	if(!err) {
		this.updateMapTopic(JSON.parse(result));
	} else {
		console.error(err);
	}
});
```
```
Error: incorrect header check
    at Zlib.zlibOnError [as onerror] (zlib.js:182:17) {
  errno: -3,
  code: 'Z_DATA_ERROR'
}
```

**Comment by deku-m** [comment]
Having same issue with getting the map into home assistant
Is there a workaround available in home assistant

**Comment by leonardpitzu** [comment]
same for me

**Comment by leonardpitzu** [comment]
I sort of patched it but CPU and RAM usage are really high (about 25% CPU and 600MB RAM on a Synology 718+ NAS)...

**Comment by Zaschii** [comment]
any news/solution?

**Comment by alexmorbo** [comment]
Any news?

**Comment by Zaschii** [comment]
Are there any news how ICantBelieveItsNotValetudo can run with 2021.01.1?

**Comment by Hypfer** [comment]
Should be fixed by #46

**Comment by mikispag** [comment]
I made further improvements [in my fork](https://github.com/mikispag/ICantBelieveItsNotValetudo).

Feel free to port it back to upstream - in my fork, I decide to remove some features I don't need.

The fork:

* greatly optimizes the algorithm generating the map (based on #33)
  + On the same hardware, with a scale factor of 2, it can compute the map in *~400ms* versus *~14000ms* (**~35x speedup**).
* avoids unnecessary computations, caching full maps and map layers
* supports map compression
* supports base-64 encoded maps
* removes support for old binary maps, supporting Valetudo 2021.01.1+ only
* removes the webserver
* removes support for overlay/underlay images
* removes support for manual cropping.

**Comment by Hypfer** [comment]
@mikispag  As far as I'm concerned, all these changes can be merged into this repo

**Comment by mikispag** [comment]
Oh, cool. I will file a PR.

**Comment by sylvaindd** [comment]
Issue was fixed in #46 
#47 is working as well.

Thanks
